public/app/plugins/panel/live/LivePanel.tsx

```diff
@@ -0,0 +1,160 @@
+import React, { PureComponent } from 'react';
+import { Unsubscribable, PartialObserver } from 'rxjs';
+import { CustomScrollbar, FeatureInfoBox, Label } from '@grafana/ui';
+import {
+  PanelProps,
+  LiveChannelStatusEvent,
+  LiveChannelAddress,
+  LiveChannel,
+  LiveChannelEvent,
+  isLiveChannelStatusEvent,
+  isLiveChannelMessageEvent,
+} from '@grafana/data';
+import { LivePanelOptions } from './types';
+import { getGrafanaLiveSrv } from '@grafana/runtime';
+
+interface Props extends PanelProps<LivePanelOptions> {}
+
+interface State {
+  error?: any;
+  channel?: LiveChannel;
+  status?: LiveChannelStatusEvent;
+  message?: any;
+}
+
+export class LivePanel extends PureComponent<Props, State> {
+  private readonly isValid: boolean;
+  subscription?: Unsubscribable;
+
+  constructor(props: Props) {
+    super(props);
+
+    this.isValid = !!getGrafanaLiveSrv();
+    this.state = {};
+  }
+
+  async componentDidMount() {
+    this.loadChannel();
+  }
+
+  componentWillUnmount() {
+    if (this.subscription) {
+      this.subscription.unsubscribe();
+    }
+  }
+
+  componentDidUpdate(prevProps: Props): void {
+    if (this.props.options?.channel !== prevProps.options?.channel) {
+      this.loadChannel();
+    }
+  }
+
+  streamObserver: PartialObserver<LiveChannelEvent> = {
+    next: (event: LiveChannelEvent) => {
+      if (isLiveChannelStatusEvent(event)) {
+        this.setState({ status: event });
+      } else if (isLiveChannelMessageEvent(event)) {
+        this.setState({ message: event.message });
+      } else {
+        console.log('ignore', event);
+      }
+    },
+  };
+
+  unsubscribe = () => {
+    if (this.subscription) {
+      this.subscription.unsubscribe();
+      this.subscription = undefined;
+    }
+  };
+
+  async loadChannel() {
+    const addr = this.props.options?.channel;
+    if (isValidChannel(addr)) {
```

Comment:
nit: if you convert the `isValidChannel` to a guard function then you don't have to do the `addr!` when fetching the channel.

Reply:
nit: another thing that might be worth considering is to invert the if statement to prevent unnecessary nesting. So what you basically could do is to is change the the ` if (isValidChannel(addr))` to `if (!isValidChannel(addr)))` and do an early return.

Reply:
nice -- thanks!